### PR TITLE
StreamVideo API tests

### PR DESCRIPTION
**Pull request by testableapple**
Resolve https://github.com/GetStream/ios-issues-tracking/issues/488

### CODE REVIEW

File: StreamVideoTests/IntegrationTests/CallCRUDTests.swift [hunk 1]
```diff
@@ -189,75 +226,186 @@ class CallCRUDTest: IntegrationTest {
...
+        var membersRequest = [MemberRequest]()
+        membersRequest.append(.init(userId: user2))
+        
+        try await call.create(members: membersRequest, memberIds: [user1])
+        let membersResponse = try await call.queryMembers()
+        
+        XCTAssertEqual(2, membersResponse.members.count)
+    }
+    
+    func test_changeTrackVisibility() async throws {
```
Comment (martinmitrevski) [inline StreamVideoTests/IntegrationTests/CallCRUDTests.swift:351]: this is only a local change, no API call is triggered

File: StreamVideoTests/IntegrationTests/CallCRUDTests.swift [hunk 2]
```diff
@@ -189,75 +226,186 @@ class CallCRUDTest: IntegrationTest {
...
+        var showTrack = await call.state.participants.first?.showTrack
+        XCTAssertEqual(showTrack, false, "Track should not be showed")
+        
+        await call.changeTrackVisibility(for: participant, isVisible: true)
+        try await customWait()
+        showTrack = await call.state.participants.first?.showTrack
+        XCTAssertEqual(showTrack, true, "Track should be showed")
+    }
+    
+    func test_updateTrackSize() async throws {
```
Comment (martinmitrevski) [inline StreamVideoTests/IntegrationTests/CallCRUDTests.swift:371]: Same here, no API request

File: StreamVideoTests/IntegrationTests/CallCRUDTests.swift [hunk 3]
```diff
@@ -189,75 +226,186 @@ class CallCRUDTest: IntegrationTest {
...
+        try await Task.sleep(nanoseconds: duration)
+    }
 }
+
+/** TODO: API tests
+ 
+- Call.swift:
+    notify()
+    accept()
+    setVideoFilter()
```
Comment (martinmitrevski) [inline StreamVideoTests/IntegrationTests/CallCRUDTests.swift:398]: you also don't need `setVideoFilter`

File: StreamVideoTests/IntegrationTests/CallCRUDTests.swift [hunk 4]
```diff
@@ -189,75 +226,186 @@ class CallCRUDTest: IntegrationTest {
...
+    currentUserCanRequestPermissions()
```
Comment (martinmitrevski) [inline StreamVideoTests/IntegrationTests/CallCRUDTests.swift:399]: This also just checks local state, shouldn't go in these tests.

File: StreamVideoTests/IntegrationTests/CallCRUDTests.swift [hunk 5]
```diff
@@ -189,75 +226,186 @@ class CallCRUDTest: IntegrationTest {
...
+    grant(permissions: )
+    grant(request: )
+    revoke()
+ 
+- StreamVideo.swift
+    makeCallsController()
```
Comment (martinmitrevski) [inline StreamVideoTests/IntegrationTests/CallCRUDTests.swift:405]: We also don't need a test for this.

File: StreamVideoTests/IntegrationTests/CallCRUDTests.swift [hunk 6]
```diff
@@ -189,75 +226,186 @@ class CallCRUDTest: IntegrationTest {
...
 
+        try await call.mute(userId: user2)
         // there is no muted event from coordinator atm
-        try await call.mute(userId: "tommaso")
     }
     
-    func test_mute_user_by_all() async throws {
-        let call = client.call(callType: "audio_room", callId: UUID().uuidString)
-        try await call.create(memberIds: ["thierry", "tommaso"])
+    // TODO: add assertion
```
Comment (martinmitrevski) [inline StreamVideoTests/IntegrationTests/CallCRUDTests.swift:310]: I guess you can fetch the participants from the session and make sure that their audio is off.

File: StreamVideoTests/IntegrationTests/CallCRUDTests.swift [hunk 7]
```diff
@@ -189,75 +226,186 @@ class CallCRUDTest: IntegrationTest {
...
-        hasVideoCapability = await tommasoCall.currentUserHasCapability(.sendVideo)
+        hasVideoCapability = await secondUserCall.currentUserHasCapability(.sendVideo)
         XCTAssertFalse(hasVideoCapability)
     }
     
-    func test_mute_user_by_id() async throws {
-        let call = client.call(callType: "audio_room", callId: UUID().uuidString)
-        try await call.create(memberIds: ["thierry", "tommaso"])
+    // TODO: add assertion
+    func test_muteUserById() async throws {
```
Comment (martinmitrevski) [inline StreamVideoTests/IntegrationTests/CallCRUDTests.swift:302]: similarly, call.get and check in the session if that user is muted.

File: StreamVideoTests/IntegrationTests/CallCRUDTests.swift [hunk 8]
```diff
@@ -189,75 +226,186 @@ class CallCRUDTest: IntegrationTest {
...
+        var actualTrackSize = await call.state.participants.first?.trackSize
+        XCTAssertEqual(expectedTrackSize, actualTrackSize)
+    }
+    
+    func customWait(nanoseconds duration: UInt64 = 2_500_000_000) async throws {
+        try await Task.sleep(nanoseconds: duration)
+    }
 }
+
+/** TODO: API tests
```
Comment (martinmitrevski) [inline StreamVideoTests/IntegrationTests/CallCRUDTests.swift:393]: Maybe keep this list internally? No need to put it in the code.

**Review by martinmitrevski — COMMENTED** [review]
Added few comments to address before merging

File: StreamVideoTests/IntegrationTests/CallCRUDTests.swift (hunk 1 above)
Comment (martinmitrevski) [reply StreamVideoTests/IntegrationTests/CallCRUDTests.swift:351]: yes, no need to test them here

File: StreamVideoTests/IntegrationTests/CallCRUDTests.swift (hunk 6 above)
Comment (testableapple) [reply StreamVideoTests/IntegrationTests/CallCRUDTests.swift]: Done 👍

File: StreamVideoTests/IntegrationTests/CallCRUDTests.swift (hunk 7 above)
Comment (testableapple) [reply StreamVideoTests/IntegrationTests/CallCRUDTests.swift:301]: Done 👍

**Review by martinmitrevski — APPROVED** [review]
great work 👏

**Pull request merged**